lib/repositories/message.js

```diff
@@ -394,6 +424,43 @@ function createApi(table, provider, sqlGenerator, dependencies) {
                 });
             });
         });
+    },
+
+    /**
+     * Deletes all messages belonging to a specific mailbox
+     */
+    removeMailbox: function remove(mailbox) {
```

Comment:
I hadn't actually thought about MWI at the time, but it seems like the obvious thing to do would be to just set both read and unread to 0 when all the messages are deleted for a mailbox.